src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/ShardedHttpExecutionServiceCluster.scala

```diff
@@ -0,0 +1,110 @@
+    val targetMember = nextAvailableMember(workflowRecs)
+    targetMember.dao.submitWorkflows(wdl, inputs, options, userInfo) map {results =>
+      (targetMember.key, results)
+    }
+  }
+
+  // following are called on a workflow that has already been submitted.
+  // therefore, we want to use the cromwell instance that has been persisted
+  // onto that workflow.
+  def status(wfe: WorkflowExecution, userInfo: UserInfo): Future[ExecutionServiceStatus] =
+    getMember(wfe).flatMap {memb => memb.status(wfe.id, userInfo)}
+
+  def callLevelMetadata(wfe: WorkflowExecution, userInfo: UserInfo): Future[ExecutionMetadata] =
+    getMember(wfe).flatMap {memb => memb.callLevelMetadata(wfe.id, userInfo)}
+
+  def outputs(wfe: WorkflowExecution, userInfo: UserInfo): Future[ExecutionServiceOutputs] =
+    getMember(wfe).flatMap {memb => memb.outputs(wfe.id, userInfo)}
+
+  def logs(wfe: WorkflowExecution, userInfo: UserInfo): Future[ExecutionServiceLogs] =
+    getMember(wfe).flatMap {memb => memb.logs(wfe.id, userInfo)}
+
+  def abort(wfe: WorkflowExecution, userInfo: UserInfo): Future[Try[ExecutionServiceStatus]] =
+    getMember(wfe).flatMap {memb => memb.abort(wfe.id, userInfo)}
+
+
+  // ====================
+  // facade-to-cluster entry points
+  // ====================
+  // for an already-submitted workflow, get the instance to which it was submitted
+  private def getMember(wfe: WorkflowExecution):Future[ExecutionServiceDAO] = wfe.executionServiceId match {
+    case Some(execId) => Future(getMember(execId))
+    case None => {
+      dataSource.inTransaction { dataAccess =>
+        dataAccess.workflowQuery.getExecutionServiceKey(wfe.id) map {execIdOption =>
+          execIdOption match {
+            case Some(execId) => getMember(execId)
+            case None => throw new RawlsException("can only process Workflow objects with an execution service key")
+          }
+        }
+      }
+    }
+  }
+
+  // for unsubmitted workflows, get the best instance to which we should submit
+  // we expect that all workflowRecs passed to this method will be a single batch;
+  // we return one target cromwell instance for the entire batch.
+  private def nextAvailableMember(workflowRecs: Seq[WorkflowRecord]): ClusterMember = {
+    // inspect the first workflow in the batch, retrieve a long value of its last-updated timestamp
+    // (could also use workflowId or some other number)
+    val shardingSeed = workflowRecs.head.statusLastChangedDate.getTime
+    val shardIndex = targetIndex(shardingSeed, members.size)
+
+    memberArray(shardIndex)
+  }
+
+
+
+  // ====================
+  // clustering methods
+  // ====================
+  private def getMember(strKey: String): ExecutionServiceDAO = getMember(ExecutionServiceId(strKey))
```

Comment:
doesn't having the convenience method defeat the purpose of having a ExecutionServiceId class? Why not just have the key be a string?